c/src/lib/libbsp/arm/shared/stm32fxxxx/i2c/hal-i2c.c

```diff
@@ -0,0 +1,331 @@
+/**
+ * @file hal-i2c.c
+ *
+ * @ingroup can
+ *
+ * @brief I2C driver for the STM32xxxx series processors. Provides a
+ * I2C driver that registers with cpukit/dev/i2c as a node in the
+ * filesystem.
+ *
+ */
+
+/*
+ * Copyright (c) 2015 Vecna Technologies, Inc.
+ *
+ * The license and distribution terms for this file may be
+ * found in the file LICENSE in this distribution or at
+ * http://www.rtems.com/license/LICENSE.
+ */
+
+//================== STMF32 Support Functions =================================
+#include <rtems.h>
+#include <hal-utils.h>
+#include <stm32f-processor-specific.h>
+#include <bspopts.h>
+
+#include stm_processor_header(TARGET_STM_PROCESSOR_PREFIX)
+#include stm_header(TARGET_STM_PROCESSOR_PREFIX, dma)
+#include stm_header(TARGET_STM_PROCESSOR_PREFIX, i2c)
+#include stm_header(TARGET_STM_PROCESSOR_PREFIX, rcc)
+#include stm_header(TARGET_STM_PROCESSOR_PREFIX, gpio)
+
+#include <hal-i2c.h>
+#include <hal-i2c-init.h>
+
+#include <math.h>
+#include <rtems/irq-extension.h>
+
+stm32_i2c_bus *bus[MAX_I2C_INSTANCES];
+I2C_HandleTypeDef I2cHandle;
+
+stm32_i2c_pin_config i2c_pin_config[MAX_I2C_INSTANCES] = {
+#if (STM32_ENABLE_I2C1)
+													{.port 		= I2C1_PORT,
+													.sck_pin 	= I2C1_SCK_PIN,
+													.sda_pin 	= I2C1_SDA_PIN,
+													.alternate_func = GPIO_AF4_I2C1},
+#endif
+#if (STM32_ENABLE_I2C2)
+													{.port 		= I2C2_PORT,
+													.sck_pin 	= I2C2_SCK_PIN,
+													.sda_pin 	= I2C2_SDA_PIN,
+													.alternate_func = GPIO_AF4_I2C2},
+#endif
+#if (STM32_ENABLE_I2C3)
+													{.port 		= I2C3_PORT,
+													.sck_pin 	= I2C3_SCK_PIN,
+													.sda_pin 	= I2C3_SDA_PIN,
+													.alternate_func = GPIO_AF4_I2C3},
+#endif
+#if (STM32_ENABLE_I2C4)
+													{.port 		= I2C4_PORT,
+													.sck_pin 	= I2C4_SCK_PIN,
+													.sda_pin 	= I2C4_SDA_PIN,
+													.alternate_func = GPIO_AF4_I2C4},
+#endif
+};
+
+const bool I2C_Enable[MAX_I2C_INSTANCES] = {
+#ifndef STM32_ENABLE_I2C1
+		false,
+#else
+		STM32_ENABLE_I2C1,
+#endif
+#ifndef STM32_ENABLE_I2C2
+		false,
+#else
+		STM32_ENABLE_I2C2,
+#endif
+#ifndef STM32_ENABLE_I2C3
+		false,
+#else
+		STM32_ENABLE_I2C3,
+#endif
+#ifndef STM32_ENABLE_I2C4
+		false,
+#else
+		STM32_ENABLE_I2C4,
+#endif
+};
+
+/**
+ * I2C Own Device Addresses
+ */
+uint32_t I2C_Address[MAX_I2C_INSTANCES] = {
+#if (STM32_ENABLE_I2C1)
```

Comment:
Likewise, if #ifdefs are used in the enumeration definition then there is no need for #else cases here.